embeddings/task-type-embedding.ipynb

```diff
@@ -0,0 +1,3079 @@
+   ],
+   "metadata": {
+    "id": "AD0zX6nu9-Wu"
+   },
+   "id": "AD0zX6nu9-Wu"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "<table align=\"left\">\n",
+    "  <td style=\"text-align: center\">\n",
+    "    <a href=\"https://colab.research.google.com/github/GoogleCloudPlatform/generative-ai/blob/main/embeddings/task-type-embedding.ipynb\">\n",
+    "      <img width=\"32px\" src=\"https://www.gstatic.com/pantheon/images/bigquery/welcome_page/colab-logo.svg\" alt=\"Google Colaboratory logo\"><br> Open in Colab\n",
+    "    </a>\n",
+    "  </td>\n",
+    "  <td style=\"text-align: center\">\n",
+    "    <a href=\"https://console.cloud.google.com/vertex-ai/colab/import/https:%2F%2Fraw.githubusercontent.com%2FGoogleCloudPlatform%2Fgenerative-ai%2Fmain%2Fembeddings%2Ftask-type-embedding.ipynb\">\n",
+    "      <img width=\"32px\" src=\"https://lh3.googleusercontent.com/JmcxdQi-qOpctIvWKgPtrzZdJJK-J3sWE1RsfjZNwshCFgE_9fULcNpuXYTilIR2hjwN\" alt=\"Google Cloud Colab Enterprise logo\"><br> Open in Colab Enterprise\n",
+    "    </a>\n",
+    "  </td>\n",
+    "  <td style=\"text-align: center\">\n",
+    "    <a href=\"https://console.cloud.google.com/vertex-ai/workbench/deploy-notebook?download_url=https://raw.githubusercontent.com/GoogleCloudPlatform/generative-ai/main/embeddings/task-type-embedding.ipynb\">\n",
+    "      <img src=\"https://www.gstatic.com/images/branding/gcpiconscolors/vertexai/v1/32px.svg\" alt=\"Vertex AI logo\"><br> Open in Vertex AI Workbench\n",
+    "    </a>\n",
+    "  </td>\n",
+    "  <td style=\"text-align: center\">\n",
+    "    <a href=\"https://github.com/GoogleCloudPlatform/generative-ai/blob/main/embeddings/task-type-embedding.ipynb\">\n",
+    "      <img width=\"32px\" src=\"https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg\" alt=\"GitHub logo\"><br> View on GitHub\n",
+    "    </a>\n",
+    "  </td>\n",
+    "</table>"
+   ],
+   "metadata": {
+    "id": "6yxMhQfFwuzQ"
+   },
+   "id": "6yxMhQfFwuzQ"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Questions and answers are not similar\n",
+    "\n",
+    "When building Retrieval Augmented Generation (RAG) systems, most designs we’ve seen use text embedding and vector search to conduct a [semantic similarity search](https://cloud.google.com/blog/products/ai-machine-learning/how-to-use-grounding-for-your-llms-with-text-embeddings). In many cases, this leads to degraded search quality, because of the \"question is not the answer\" problem. For example, a question like \"Why is the sky blue?\" and its answer, \"The scattering of sunlight causes the blue color\", have distinctly different meanings as separate statements. Likewise, if you ask \"What's the best birthday present for my kid?\" to a RAG system with a similarity search, it would be hard to find items like \"Nintendo Switch\" or \"Lego sets\", as those answers are not semantically similar to the question.\n",
```

Comment:
```suggestion
    "When building Retrieval Augmented Generation (RAG) systems, most designs we've seen use text embedding and vector search to conduct a [semantic similarity search](https://cloud.google.com/blog/products/ai-machine-learning/how-to-use-grounding-for-your-llms-with-text-embeddings). In many cases, this leads to degraded search quality, because of the \"question is not the answer\" problem. For example, a question like \"Why is the sky blue?\" and its answer, \"The scattering of sunlight causes the blue color\", have distinctly different meanings as separate statements. Likewise, if you ask \"What's the best birthday present for my kid?\" to a RAG system with a similarity search, it would be hard to find items like \"Nintendo Switch\" or \"Lego sets\", as those answers are not semantically similar to the question.\n",
```